### ISSUE TITLE
Finalize exported serialization package

### ISSUE THREAD

**Issue by sagikazarmark** [lstrojny/fxmlrpc#66]
I would like to finalize this package soon:

https://github.com/fxmlrpc/serialization

What do you think about integrating it into this library afterwards? (extend/composition)

**Comment by sagikazarmark** [comment]
Ping @lstrojny

**Comment by lstrojny** [comment]
@sagikazarmark In general I believe that’s a good idea. What I disagree with is the B/C breaks it would bring in it’s current state. The breaks would mostly be changes in interface naming and the namespace. I don’t think those are necessary and would make life just harder for the users of this package, especially since we had quite a bit of breaking changes in the past already, so I want to give them some rest. A smaller issue are the whitespace changes that from my point of view make readability worse.

**Comment by sagikazarmark** [comment]
Sorry, I probably wasn't clear: I would like to integrate it in a non-BC breaking way if that's possible. Now that I kind of finalized the extracted codebase, it would probably be better to avoid maintaining the same code in two different libraries.

**Comment by sagikazarmark** [comment]
So probably the best way would be to have some kind of composition of the new serializers in the "old" ones.

**Comment by lstrojny** [comment]
@sagikazarmark yep, that sounds good. I don't want to break B/C for the client at all right now. But including the serializers as an external package instead, sure.

**Comment by sagikazarmark** [comment]
Okay, I will try to pick up this one.

BTW Are you still committed to the separation work we started a few years ago @lstrojny? Because if I can find the time I will try to move things forward a little bit.

**Comment by lstrojny** [comment]
@sagikazarmark happy to continue the separation although I don’t expect to have too much time to spend on it.
